internal/types/types.go

```diff
@@ -1700,3 +1701,30 @@ type Webhook struct {
 	CreatedByUserID int32
 	UpdatedByUserID int32
 }
+
+// CodeHostURN is an identifier for a unique code host in the form
+// of its host URL.
+type CodeHostURN struct {
+	url *url.URL
+}
+
+// ParseCodeHostURN takes a code host URL string, normalizes it,
+// and returns a CodeHostURN. If the string is required, use the
+// .String() method on the CodeHostURN.
+func ParseCodeHostURN(urnString string) (CodeHostURN, error) {
+	url, err := url.Parse(urnString)
+	if err != nil {
+		return CodeHostURN{}, err
```

Comment:
```suggestion
		return nil, err
```

I think returning `nil` here is fine

Reply:
Then you have to return a pointer to `CodeHostURN` instead which I don't like :( feels unnecessary

Reply:
Yeah, I don't think you need a pointer here.